src/configure/helper/commonHelper.ts

```diff
@@ -63,4 +67,36 @@ export function stringCompareFunction(a: string, b: string): number {
         return 1;
     }
     return 0;
+}
+
+export async function executeFunctionWithRetry(func: () => Promise<any>, retryCount: number = 100, retryTimeoutInSec: number = 300, retryIntervalTimeInSec: number = 2, errorMessage?: string): Promise<any> {
+    let internalErrorMessage = '';
+    let functionExecutionPromise = Q.Promise(async (resolve, reject) => {
+        let result = null;
+        for (;retryCount > 0; retryCount--) {
+            try {
+                result = await func();
+                resolve(result);
+                break;
+            }
+            catch (error) {
+                internalErrorMessage = errorMessage ? errorMessage.concat(' Internal Error: ', JSON.stringify(error)) : JSON.stringify(error);
```

Comment:
do we need to concat all error messages. You can log messages for each retry in console and throw only the last error message.

Reply:
This line only concats the last error message with the error message sent by the caller.
So it will work in this way 
1. errorMessage (sent by caller) -> 'Failed while assigning role'
2. error (thrown from api call) -> { 'message':'You do not have permission to assign role', error: type, stack: stack}

internalErrorMessage -> 'Failed while assigning role'.concat(' Internal Error: ' , Json.Stringify(error))
final result will be the above. As we always update internalErrorMessage only, which is always only assigned and not used in concat.


I can log all retry error messages though

Reply:
But logging all retry error's might be overwhelming, what do you say ?

We are getting the final error message which gets logged against a correct tracepoint, If we want to log all error logs for retry then we either need a common trace point or need to pass appropriate tracepoint

Reply:
But then you don't need to update/maintain internalErrorMessage. you can directly check retryCount ==0 and then throw

```
if (retryCount <= 0) {
            reject(errorMessage.concat(JSON.stringify(error)));
}
```